tests/functions.php

```diff
@@ -0,0 +1,111 @@
+<?php
+require '../functions.php';
+use PHPUnit\Framework\TestCase;
+use PHPUnit\Framework\Error;
+
+class Functions extends TestCase
+{
+
+    public function testSuccess1_displayFilmsAndRoles()
+    {
+        $inputFilms = [
+            [ 'id'=>  1, 'title' => 'The Arms Dealer', 'year_produced' =>  '2018-10-01', 'type' => 'horror' ],
+            [ 'id'=>  2, 'title' => 'Buddy\'s Budget Hitman Service', 'year_produced' =>  '2019-08-20', 'type' => 'drama' ]
+        ];
+        $inputFilmRoles = [
+            ["film-id"=> "1", "name"=>  "producer", "id"=> 1],
+            ["film-id"=> "1", "name"=>  "writer", "id"=> 2],
+            ["film-id"=> "1", "name"=>  "camera operator", "id"=> 6]
+        ];
+        $expectedOutput = '<article class="container__film"><h2>Film: The Arms Dealer</h2><p>Year Produced: 2018-10-01</p><p>Genre: horror</p><p>My Roles: </p><ul><li>producer</li><li>writer</li><li>camera operator</li></ul></article><article class="container__film"><h2>Film: Buddy\'s Budget Hitman Service</h2><p>Year Produced: 2019-08-20</p><p>Genre: drama</p><p>My Roles: </p><ul></ul></article>';
+        $result = displayFilmsAndRoles($inputFilms, $inputFilmRoles);
+        $this->assertEquals($expectedOutput, $result);
+    }
+
+    /**
+     * Tests for Failure if the required field 'id' is spelled incorrectly ie doesnt exist - if other fields are mis-spelled (eg year or type) then their value is shown as blank, as a film without an id or title should not be shown at all
+     */
+    public function testFailure1_displayFilmsAndRolesWrongId()
```

Comment:
Maybe just add the word key on the end of the names as it's then explicit that the keys are wrong not the values.